### ISSUE TITLE
Record type referencing does not work in jBallerina

### ISSUE THREAD

**Issue by Kishanthan** [ballerina-platform/ballerina-lang#15780]
```ballerina
type Person record {
    string name;
    int age;
};

type Student record {
    *Person;
    float gpa;
};
```

Above doesn't work and jvm compiler backend fails with the an error.

```
error: cannot find function definition for : <init> {}
	at compiler_backend_jvm:0.0.0:isBIRFunctionExtern(jvm_package_gen.bal:78)
	   compiler_backend_jvm:0.0.0:genStaticCall(jvm_terminator_gen.bal:330)
	   compiler_backend_jvm:0.0.0:genBuiltinTypeAttachedFuncCall(jvm_terminator_gen.bal:322)
	   compiler_backend_jvm:0.0.0:genCall(jvm_terminator_gen.bal:308)
	   compiler_backend_jvm:0.0.0:genCallTerm(jvm_terminator_gen.bal:244)
	   compiler_backend_jvm:0.0.0:genTerminator(jvm_terminator_gen.bal:44)
	   compiler_backend_jvm:0.0.0:generateBasicBlocks(jvm_method_gen.bal:578)
	   compiler_backend_jvm:0.0.0:generateMethod(jvm_method_gen.bal:192)
	   compiler_backend_jvm:0.0.0:createRecordMethods(jvm_value_gen.bal:364)
	   compiler_backend_jvm:0.0.0:createRecordValueClass(jvm_value_gen.bal:353)
	   compiler_backend_jvm:0.0.0:generateValueClasses(jvm_value_gen.bal:44)
	   compiler_backend_jvm:0.0.0:generatePackage(jvm_package_gen.bal:128)
	   compiler_backend_jvm:0.0.0:generateJarBinary(main.bal:50)
```

**Comment by warunalakshitha** [comment]
Once fix the issue lets revert the change in  #15782 in tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/type-guard.bal file.

**Comment by SupunS** [comment]
Object Referencing currently works, but does not copy the default-values of the expressions.

Need to figure out a way to get the default values..
